chromium_src/components/crx_file/crx_creator.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * you can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/crx_file/crx_creator.h"
+
+namespace crx_file {
+
+class CrxFileHeader;
+
+std::string GetCrxId_BraveImpl(const std::string& key, CrxFileHeader* header);
+
+}  // namespace crx_file
+
+#include "src/components/crx_file/crx_creator.cc"
+
+namespace crx_file {
+
+// Override for GetCrxId() in SignArchiveAndCreateHeader() to generate the
+// correct signed data for the second signature.
+std::string GetCrxId_BraveImpl(const std::string& key, CrxFileHeader* header) {
+  if (header->sha256_with_rsa_size() > 0) {
+    const AsymmetricKeyProof& first_proof = header->sha256_with_rsa()[0];
+    return GetCrxId(first_proof.public_key());
+  }
+  return GetCrxId(key);
+}
+
+CreatorResult CreateWithPublisherKey(const base::FilePath& output_path,
+                                     const base::FilePath& zip_path,
+                                     crypto::RSAPrivateKey* signing_key,
```

Comment:
It might be helpful here to rename `signing_key` to `developer_key` (or something similar) since `publisher_key` is also a signing key.

Reply:
Renamed to `developer_key`